Babylon/commands/abba/check.py

```diff
@@ -0,0 +1,103 @@
+from logging import getLogger
+from click import File, argument, command
+import pandas as pd
+import plotly.offline as pyo
+import plotly.graph_objs as go
+import plotly.express as px
+from pyparsing import Any
+from Babylon.commands.abba.common import dataframe_to_dict
+from Babylon.utils.credentials import pass_azure_token
+from Babylon.utils.decorators import injectcontext, retrieve_state
+from Babylon.utils.response import CommandResponse
+from Babylon.commands.api.scenarioruns.services.api import ScenarioRunService
+
+logger = getLogger("Babylon")
+
+
+def get_scenariorun_status(service_state, azure_token):
+    # get scenariorun status
+    service = ScenarioRunService(state=service_state, azure_token=azure_token)
+    response = service.status()
+    if response is None:
+        logger.error(f"Failed to get status of {service_state['api']['scenariorun_id']}")
+    return response.json()
+
+
+def summarize(data: dict, i: int):
+    df = pd.DataFrame.from_records(data['nodes'])[['containerName', 'startTime', 'endTime']]
+    df['startTime'] = pd.to_datetime(df['startTime'])
+    df['endTime'] = pd.to_datetime(df['endTime'])
+    df['duration'] = (df['endTime'] - df['startTime']).dt.total_seconds()
+    logger.info(f"Report {i} generated")
+    return df
+
+
+def generate_report(summary_df, detailed):
+    """Generate summary report from simulation run data."""
+    overall_plot = go.Figure(data=[go.Bar(x=summary_df['duration'], y=summary_df['scenariorunId'], orientation='h')])
+    overall_plot.update_layout(title='Scenariorun Durations')
+    top = pyo.plot(overall_plot, include_plotlyjs=False, output_type='div')
+    fig_list = []
+    for i, detailed_data in enumerate(detailed):
+        fig = px.timeline(detailed_data,
+                          title=f"Detailed report for {'i'}",
+                          x_start="startTime",
+                          x_end="endTime",
+                          y="containerName")
+        fig.update_traces(text=detailed_data['duration'], textposition='outside')
+        fig.update_layout(title="Execution time by step (seconds)", xaxis_title="Time", yaxis_title="Step")
+        fig_list.append(pyo.plot(fig, include_plotlyjs=False, output_type='div'))
+    # Generate static HTML page (should be viewed online)
+    html_content = f"""
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>Simulation Report</title>
+  </head>
+  <body>
+    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+    <h2>Total execution time</h2>
+    {top}
+    <h2>Execution time by step</h2>
+    {"".join(fig_list)}
+  </body>
+  </html>
+  """
+    with open("simulation_report.html", "w") as file:
+        file.write(html_content)
+
+
+@command()
+@injectcontext()
+@pass_azure_token("csm_api")
+@retrieve_state
+@argument("input", type=File('r'))
+@argument("var_types", type=File('r'))
+def check(state: Any, azure_token: str, input: str, var_types: str) -> CommandResponse:
```

Comment:
Maybe a more explicit command's name, 'check_run_status'?

Reply:
since this is a command in the group abba I thought it might be better to keep it short. The user will be typing `babylon abbba run check` to check the run of abba.